modules/auxiliary/gather/query_ldap.rb

```diff
@@ -0,0 +1,279 @@
+      Opt::RPORT(389), # Set to 636 for SSL/TLS
+      OptEnum.new('OUTPUT_FORMAT', [true, 'The output format to use', 'table', ['csv', 'table', 'json']]),
+      OptString.new('BASE_DN', [false, 'LDAP base DN if you already have it']),
+      OptString.new('QUERY_FILE_PATH', [false, 'Path to the JSON or YAML file to load and run queries from'], conditions: %w[ACTION == RUN_QUERY_FILE])
+    ])
+  end
+
+  def perform_ldap_query(ldap, filter, attributes)
+    returned_entries = ldap.search(base: @base_dn, filter: filter, attributes: attributes)
+    query_result = ldap.as_json['result']['ldap_result']
+    case query_result['resultCode']
+    when 0
+      vprint_good('Successfully queried LDAP server!')
+    when 1
+      print_error("Could not perform query #{filter}. Its likely the query requires authentication.")
+      print_error(query_result['errorMessage'])
+    else
+      print_error("Query #{filter} failed with error: #{query_result['errorMessage']}")
+    end
+    if returned_entries.nil? || returned_entries.empty?
+      print_error("No results found for #{filter}.")
+      nil
+    else
+      returned_entries
+    end
+  end
+
+  def generate_rex_tables(entries, format)
+    entries.each do |entry|
+      tbl = Rex::Text::Table.new(
+        'Header' => entry['dn'][0].split(',').join(' '),
+        'Indent' => 1,
+        'Columns' => ['Name', 'Attributes']
+      )
+
+      for attr in entry.attribute_names
+        if format == 'table'
+          tbl << [attr, entry[attr].join(' || ')] unless attr == :dn # Skip over DN entries for tables since DN information is shown in header.
+        else
+          tbl << [attr, entry[attr].join(' || ')] # DN information is not shown in CSV output as a header so keep DN entries in.
+        end
+      end
+
+      case format
+      when 'table'
+        print_status(tbl.to_s)
+      when 'csv'
+        print_status(tbl.to_csv)
+      else
+        print_error("Invalid format #{format} passed to generate_rex_tables!")
+        break
+      end
+    end
+  end
+
+  def output_json_data(entries, _columns)
+    entries.each do |entry|
+      result = ''
+      data = {}
+      for attr in entry.attribute_names
+        data[attr] = entry[attr].join(' || ')
+      end
+      result << JSON.pretty_generate(data) + ",\n"
+      result.gsub!(/},\n$/, '}')
+      print_status(entry['dn'][0].split(',').join(' '))
+      print_line(result)
+    end
+  end
+
+  def output_data_table(entries, _columns)
+    generate_rex_tables(entries, 'table')
+  end
+
+  def output_data_csv(entries, _columns)
+    generate_rex_tables(entries, 'csv')
+  end
+
+  def perform_multiple_queries_from_file(ldap, parsed_file)
+    parsed_file['queries'].each do |query|
+      unless query['name'] && query['filter'] && query['columns']
+        print_error("Each query in the query file must at least contain a 'name', 'filter' and 'columns' attribute!")
+        break
+      end
+      attributes = query['columns']
```

Comment:
```suggestion
      attributes = query['attributes']
```

Reply:
Fixed in https://github.com/rapid7/metasploit-framework/pull/16598/commits/cca391757323058c5b6f8fcebe1e178ce21d3f97